lib/gst/player/gstplayer.c

```diff
@@ -3179,6 +3180,84 @@ gst_player_set_visualization_enabled (GstPlayer * self, gboolean enabled)
       enabled ? "Enabled" : "Disabled");
 }
 
+/**
+ * gst_player_get_color_balance_list_channels:
+ * @player: #GstPlayer instance
+ *
+ * Retrieve a list of the available channels.
+ *
+ * Returns: (element-type Gst.ColorBalanceChannel) (transfer none): A
+ *          GList containing pointers to #GstColorBalanceChannel
+ *          objects. The list is owned by the #GstColorBalance
+ *          instance and must not be freed.
+ */
+const GList *
+gst_player_get_color_balance_list_channels (GstPlayer * self)
+{
+  g_return_val_if_fail (GST_IS_PLAYER (self), NULL);
+
+  if (!GST_IS_COLOR_BALANCE (self->playbin))
+    return NULL;
+
+  return gst_color_balance_list_channels (GST_COLOR_BALANCE (self->playbin));
+}
+
+/**
+ * gst_player_set_color_balance_value:
+ * @player: #GstPlayer instance
+ * @channel: #GstColorBalanceChannel instance
+ * @value: The new value for the channel
+ *
+ * Sets the current value of the channel to the passed value, which must
+ * be between min_value and max_value.
+ *
+ * See Also: The #GstColorBalanceChannel.min_value and
+ *         #GstColorBalanceChannel.max_value members of the
+ *         #GstColorBalanceChannel object.
+ */
+void
+gst_player_set_color_balance_value (GstPlayer * self,
+    GstColorBalanceChannel * channel, gint value)
+{
+  g_return_if_fail (GST_IS_PLAYER (self));
+  g_return_if_fail (channel);
+
+  if (!GST_IS_COLOR_BALANCE (self->playbin))
+    return;
+
+  gst_color_balance_set_value (GST_COLOR_BALANCE (self->playbin), channel,
+      value);
+}
+
+/**
+ * gst_player_get_color_balance_value:
+ * @player: #GstPlayer instance
+ * @channel: #GstColorBalanceChannel instance
+ *
+ * Retrieve the current value of the indicated channel, between min_value
+ * and max_value.
+ *
+ * See Also: The #GstColorBalanceChannel.min_value and
+ *         #GstColorBalanceChannel.max_value members of the
+ *         #GstColorBalanceChannel object.
+ *
+ * Returns: The current value of the channel.
+ */
+gint
```

Comment:
Not sure if these plain gint values are useful here. We should at least document their range, and make sure it's always the same range per channel.


I think in general I would prefer if we don't expose the GstColorBalance API in GstPlayer, but instead have a simple API around an enum { BRIGHTNESS, CONTRAST, SATURATION, HUE } for setting things directly